test/CMakeLists.txt

```diff
@@ -8,33 +15,37 @@
 # It also lets running the tests much faster.
 if (APPLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ffp-contract=fast")
-endif()  
+endif()
 
 function(boost_geometry_add_unit_test prefix item)
   set(unit_test_name "boost_geometry_${prefix}_${item}")
   add_executable(${unit_test_name} ${item}.cpp)
 
-  # Add a dependendcy to Boost.Geometry
-  target_link_libraries(${unit_test_name} 
+  # Add a dependency to Boost.Geometry
+  target_link_libraries(${unit_test_name}
     PRIVATE
       Boost::geometry)
 
   # For unit tests, add a dependency to the unit test framework (in header only mode)
-  target_link_libraries(${unit_test_name} 
+  target_link_libraries(${unit_test_name}
     PRIVATE
       Boost::included_unit_test_framework)
-  
+
   # Include the main Geometry test folder and the current folder
   target_include_directories(${unit_test_name}
     PRIVATE
-      "${PROJECT_SOURCE_DIR}/test" 
+      "${PROJECT_SOURCE_DIR}/test"
+      "${PROJECT_SOURCE_DIR}/index/test"
       .)
 
   # To compile with C++14
   target_compile_features(${unit_test_name} PRIVATE cxx_std_14)
```

Comment:
The PR description mentions: `test compilation with C++17` 
But won't this setting override that?

Reply:
It seems that cmake selects the highest version of the standard specified in both CMAKE_CXX_STANDARD and target_compile_features.
see also https://discourse.cmake.org/t/how-to-handle-different-cxx-standards-in-the-same-project/6510/6

However, it is not very neat to mix CMAKE_CXX_STANDARD and target_compile_features. As the code is right now I think it is simpler to just use CMAKE_CXX_STANDARD and avoid target_compile_features, as the former only needed to be set once while the latter need to be set for each individual test. But maybe for this PR it is OK as is. What do you think?

Reply:
I agree, maybe we should just use `CMAKE_CXX_STANDARD`
And for this PR, it is OK as it is.

Agreed.